packages/react-components/src/organisms/DiscussionModal.tsx

```diff
@@ -53,27 +53,42 @@ const dismissButtonStyles = css({
   },
 });
 
-type QuickCheckReplyModalProps = {
+type DiscussionModalProps = {
+  title: string;
+  editorLabel: string;
+  discussionType: DiscussionType;
+  ruleMessage: string;
   onDismiss: () => void;
   discussionId: string;
-  onReplyToDiscussion: (
+  onSave: (
     id: string,
-    patch: DiscussionPatchRequest,
+    data: DiscussionPatchRequest | DiscussionCreateData,
   ) => Promise<void>;
 };
 
-type QuickCheckReplyModalData = {
-  replyText: string;
-};
-const QuickCheckReplyModal: React.FC<QuickCheckReplyModalProps> = ({
-  onDismiss,
+type DiscussionModalData =
+  | {
+      replyText: string;
+    }
+  | {
+      message: string;
+    };
```

Comment:
I was thinking about a way to make this simpler, I think we could make this change 
<img width="1477" alt="Screenshot 2024-12-18 at 06 33 02" src="https://github.com/user-attachments/assets/d3ed533a-893d-4676-a8ad-baa53f316d53" />
in `apps/crn-frontend/src/network/teams/api.ts`

and this change

<img width="1663" alt="Screenshot 2024-12-18 at 06 37 22" src="https://github.com/user-attachments/assets/4307b5f1-26fb-4373-86de-760dadb3d655" />

in `apps/crn-frontend/src/network/teams/state.ts`

Then we could have 
```suggestion
type DiscussionModalData = {
  text: string;
};
```